contrib/epee/src/net_ssl.cpp

```diff
@@ -641,6 +641,54 @@ bool ssl_options_t::handshake(
   return true;
 }
 
+std::string get_hr_ssl_fingerprint(const X509 *cert, const EVP_MD *fdig)
+{
+  unsigned int j;
+  unsigned int n;
+  unsigned char md[EVP_MAX_MD_SIZE];
+  std::string fingerprint;
+
+  if (!X509_digest(cert, fdig, md, &n))
```

Comment:
Check for `cert` and `fdig` against `nullptr`? Or does openssl do this check already (this is already in the codebase like this, so)..

Reply:
Yeah that can't hurt...

Reply:
Done!